### PR TITLE
Feature/s4 heads up notifications

### PR DESCRIPTION
**Pull request by federicoboschini**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -23,10 +24,27 @@
...
-        <activity android:name=".FeedbackPlainActivity"
-            android:label="Feedback activity"/>
+        <activity
+            android:name=".CouponPlainActivity"
+            android:label="Coupon detail activity" />
+        <activity
+            android:name=".FeedbackPlainActivity"
+            android:label="Feedback activity" />
+
+        <!--  Built-in NearIT-UI Intent service, replace with your own if you want to handle notifications -->
```
Comment (catt-stefano) [inline app/src/main/AndroidManifest.xml]: Add "For info, read the documentation section about (...)"

File: app/src/main/java/com/nearit/ui_bindings_sample/NearItUISample.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+/**
+ * Created by Federico Boschini on 22/09/17.
+ */
+
+public class NearItUISample extends Application {
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        NearITUIBindings.enableAutomaticForegroundNotifications(this);
```
Comment (catt-stefano) [inline app/src/main/java/com/nearit/ui_bindings_sample/NearItUISample.java]: Add a comment before the enable statement

File: docs/NOTIFICATIONS.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# NearIt-UI for notifications
+### Heads-up notifications for in-app content
+NearIt-UI can manage and show in-app content (read more about background and foreground notifications [here](http://nearit-android.readthedocs.io/en/latest/in-app-content/))
```
Comment (catt-stefano) [inline docs/NOTIFICATIONS.md]: "NearIt-UI can be set to manage and show notifications for foreground content (read more...)"

File: docs/NOTIFICATIONS.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+for you. When a user enters in a specific beacon range, an heads-up notification (on devices with Android API level 21+) will show. If the user clicks on it, the right dialog will appear and the right tracking events will be handled.
```
Comment (catt-stefano) [inline docs/NOTIFICATIONS.md]: Foreground content will be delivered through an heads-up system notification. If the user ... handled.

File: docs/NOTIFICATIONS.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+![ranging_notif](notifications.gif)
+
```
Comment (catt-stefano) [inline docs/NOTIFICATIONS.md]: Note: the heads-up notification overlay is only available for API level 21+. On older devices the notification will still appear, but with no overlay.

File: docs/NOTIFICATIONS.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        //  ...
+    }
+}
+```
+
+Please note that foreground notifications will auto-dismiss after a small amount of time.
+
+### Background notifications
+
+NearIT-SDK has a built-in service that will handle every background event and show simple notifications.
```
Comment (catt-stefano) [inline docs/NOTIFICATIONS.md]: Tapping on those notification will deliver an Intent to your app launcher.

File: docs/NOTIFICATIONS.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+If you want to directly show contents with the convenient UIs when a notification is tapped, you have to add the `NearItUIIntentService` to the manifest of your application.
```
Comment (catt-stefano) [inline docs/NOTIFICATIONS.md]: If you prefer to let the NearIt-UI library handle the tap and immediately show the convininent UIs when a....

**Review by catt-stefano — CHANGES_REQUESTED** [review]
Still haven't tested it manually.

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (federicoboschini) [reply app/src/main/AndroidManifest.xml]: @catt-stefano should I refer to the SDK documentation?

File: app/src/main/java/com/nearit/ui_bindings_sample/NearItUISample.java (hunk 2 above)
Comment (federicoboschini) [reply app/src/main/java/com/nearit/ui_bindings_sample/NearItUISample.java]: done

File: docs/NOTIFICATIONS.md (hunk 3 above)
Comment (federicoboschini) [reply docs/NOTIFICATIONS.md]: done

File: docs/NOTIFICATIONS.md (hunk 4 above)
Comment (federicoboschini) [reply docs/NOTIFICATIONS.md]: done

File: docs/NOTIFICATIONS.md (hunk 5 above)
Comment (federicoboschini) [reply docs/NOTIFICATIONS.md]: done

File: docs/NOTIFICATIONS.md (hunk 6 above)
Comment (federicoboschini) [reply docs/NOTIFICATIONS.md]: done

File: docs/NOTIFICATIONS.md (hunk 7 above)
Comment (federicoboschini) [reply docs/NOTIFICATIONS.md]: done

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (catt-stefano) [reply app/src/main/AndroidManifest.xml]: You should refer to the NearIT-UI documentation regarding this particular feature, maybe not by link, but at least give enough info to let a user know that this is for a certain feature and it's not strictly required

File: docs/NOTIFICATIONS.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        <category android:name="android.intent.category.DEFAULT" />
+    </intent-filter>
+    <intent-filter>
+        <action android:name="it.near.sdk.permission.GEO_MESSAGE" />
+        <category android:name="android.intent.category.DEFAULT" />
+    </intent-filter>
+</service>
+```
+
+If your app is in background (e.g. an user closed it and maybe killed it) your user will receive a simple notification. When the notification is tapped, the app is launched and the content dialog appears.
```
Comment (catt-stefano) [inline docs/NOTIFICATIONS.md]: When your app is in background...

File: docs/NOTIFICATIONS.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+When your app is in foreground, background events are notified in heads-up style.
```
Comment (catt-stefano) [inline docs/NOTIFICATIONS.md]: Additionally, when your app is in foreground, ...

**Review by catt-stefano — CHANGES_REQUESTED** [review]

**Pull request merged**